src/shims/unix/fd.rs

```diff
@@ -330,6 +325,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
 
         Ok(Scalar::from_i32(if let Some(file_descriptor) = this.machine.fds.remove(fd) {
             let result = file_descriptor.close(this.machine.communicate())?;
+            // return `0` if close is successful
+            let result = result.map(|_| 0i32);
```

Comment:
```suggestion
            let result = result.map(|()| 0i32);
```
This makes sure we notice if the argument type becomes something else that we do not want to throw away